### ISSUE TITLE
[css-syntax-3] Allow a descriptor matching a property to cascade and get a priority

### ISSUE THREAD

**Issue by cdoublev** [w3c/csswg-drafts#7838]
> Within a `<style-block>` or `<declaration-list>`, `!important` is automatically invalid on any descriptors. If the rule accepts properties, the spec for the rule must define whether the properties interact with the cascade, and with what specificity. If they don’t interact with the cascade, properties containing `!important` are automatically invalid; otherwise using `!important` is valid and causes the declaration to be important for the purposes of the cascade.

According to https://github.com/w3c/csswg-drafts/issues/5969#issuecomment-776373782, all declarations in `@page` are for descriptors. Therefore I assume the above (from [CSS Syntax](https://w3c.github.io/csswg-drafts/css-syntax-3/)) also applies to *descriptors matching a property*. If so, the first sentence seems confusing to me.

> Declarations in page and margin contexts cascade just like declarations in style rule for elements.

https://w3c.github.io/csswg-drafts/css-page-3/#cascading-and-page-context

**Comment by fantasai** [comment]
Hmm, this seems to me like terminology just failing to link up correctly. Syntax seems to think the declarations inside @page are properties, and Page thinks they're descriptors.

**Comment by cdoublev** [comment]
Yeah, imo, descriptors matching a property should be defined as functionnaly equivalent to properties, ie. their declarations cascade if the spec for the rules accepting them defines it, and if so, they also allow CSS-wide keywords and custom variables, noting that `@page` and margin rules only accept `inherit` though.

**Comment by cdoublev** [comment]
Changing the title to broaden the topic because it is also unclear if descriptor declarations accept custom variables and CSS-wide keywords. At least, Chrome/FF accept them in some cases.

Below, in Chrome/FF: 

  - declarations for `font-display` are all parsed as invalid
  - declarations for `size` are all parsed as valid
  - declarations for `color` are all parsed as valid in Chrome / invalid in FF (bug imo, even with `inherit`)

```css
@font-face { 
  font-display: block !important; 
}
@font-face { 
  font-display: initial; 
}
@font-face { 
  font-display: var(--custom); 
}
@page { 
  color: green !important; 
  size: 1px !important; 
}
@page { 
  color: initial; 
  size: initial; 
}
@page { 
  color: var(--custom); 
  size: var(--custom); 
}
```

  > The value of a custom property can be substituted into the value of another property with the var() function.

https://drafts.csswg.org/css-variables-1/#using-variables

  > Several CSS-wide property values are defined below; declaring a property to have these values explicitly specifies a particular defaulting behavior. As specified in CSS Values and Units [css-values-3], all CSS properties can accept these values.

https://drafts.csswg.org/css-cascade-5/#defaulting-keywords

In my opinion:

  - custom variables should always be valid, for all type of descriptors
  - all CSS-wide keywords should be valid if declarations cascade, for all type of descriptors
  - `initial` should always be valid, even for "pure" descriptors
